components/telemetry-operator/internal/secret/secret_helper.go

```diff
@@ -4,7 +4,6 @@ import (
 	"bytes"
 	"context"
 	"fmt"
-	"strings"
```

Comment:
All good with the changes in your PR, but on line 59 we log incorrect information: 
```
log.Error(err, "Failed reading secret '%s' from namespace '%s'", secretKey.Name, secretKey.Namespace)
```

`log.Error` expects an error, error message and then an arbitrary amount of key value pairs. Our current implementation assumes that the function formats the error message, which is not the case.

Just noticed this last time and thought it may be worth fixing it before the release.